File: problem-solving/_lc-58-easy-length-of-last-word/1-solution.py
"""
https://leetcode.com/problems/length-of-last-word/

Complexity
----------
Time: O(n)
Space: O(1)

n = length of whole string
"""

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        l = 0
        n = len(s) - 1
        while n >= 0 and s[n] == ' ':
            n -= 1
        while n >= 0 and s[n] != ' ':
            n -= 1
            l += 1
        return l

File: problem-solving/_lc-58-easy-length-of-last-word/1-solution.go
/**
 * https://leetcode.com/problems/length-of-last-word/
 * 
 * Complexity
 * ----------
 * Time: O(n)
 * Space: O(1)
 * 
 * n = length of whole string
 */

func lengthOfLastWord(s string) int {
    count := 0
    for i := len(s) - 1; i >= 0; i-- {
        if s[i] != ' ' {
            count += 1
        } else if count > 0 {
            return count
        }
    }
    return count
}
